tests/unit/package_managers/test_general.py

```diff
@@ -132,3 +141,84 @@ def test_extract_git_info(url: str, nonstandard_info: Any) -> None:
     }
     info.update(nonstandard_info or {})
     assert general.extract_git_info(url) == info
+
+
+@pytest.mark.asyncio
+async def test_async_download_binary_file(tmp_path: Path, caplog: pytest.LogCaptureFixture) -> None:
+    url = "http://example.com/file.tar"
+    download_path = tmp_path / "file.tar"
+    chunk_size = 8192
+
+    # Create mock response and session
+    response, session = MagicMock(), MagicMock()
+    response.content.read = MockReadChunk().read_chunk
+
+    async def mock_aenter() -> MagicMock:
+        return response
+
+    session.get().__aenter__.side_effect = mock_aenter
+
+    await _async_download_binary_file(session, url, download_path, chunk_size=chunk_size)
+
+    # Verify that the file was downloaded correctly
+    with open(download_path, "rb") as f:
+        assert f.read() == b"first_chunk-second_chunk-"
+
+    # # Verify that session.get was called with the correct arguments
+    assert session.get.called
+    assert session.get.call_args == mock.call(url, auth=None, raise_for_status=True)
+
+    # Test unsuccessful download with an exception
+    exception_message = "This is a test exception message."
+    session.get().__aenter__.side_effect = Exception(exception_message)
+
+    with pytest.raises(Exception) as exc_info:
```

Comment:
Minor nitpick: it would be better to split the exception tests into a separate test case, or parameterize it (unless it makes things too complicated)

Reply:
Agreed, it's much better to have one test -> one scenario or a parameterized test (don't forget the 'id' arg :grin:)